test/e2e/serial/tests/workload_placement.go

```diff
@@ -600,6 +901,32 @@ func availableResourceType(nrtInfo nrtv1alpha1.NodeResourceTopology, resName cor
 	return res.DeepCopy()
 }
 
+func leastAvailableResourceQtyInAllZone(nrtInfo nrtv1alpha1.NodeResourceTopology, baseload baseload.Load, resName corev1.ResourceName) resource.Quantity {
+	var res, zeroVal resource.Quantity
+
+	// There is no way to exactly determine how the baseload is ditributed across NUMA nodes
+	// so we subtract baseload from both the NUMA nodes to be on the safe side.
+	for _, zone := range nrtInfo.Zones {
+		zoneQty, ok := e2enrt.FindResourceAvailableByName(zone.Resources, resName.String())
+		if !ok {
+			continue
+		}
+
+		switch resName {
+		case corev1.ResourceCPU:
+			zoneQty.Sub(baseload.CPU)
+
+		case corev1.ResourceMemory:
+			zoneQty.Sub(baseload.Memory)
+		}
+		if zoneQty.IsZero() || zoneQty.Cmp(zeroVal) < 0 {
+			zoneQty = res
+		}
+		res = zoneQty.DeepCopy()
+	}
+	return res
```

Comment:
the name confuses me, can you tell please what's the goal of this function?

Reply:
Sure, I am open to changing the name to something more descriptive. I added a comment to explain the goal of the function and refactored it as well.
Just to summarize: we are trying to admit both the replicas of deployment on both NUMA nodes and need to determine the least available resource out of both the NUMA nodes. Here is what mean with an example:
Assuming resources after deducting baseload are:
NUMA 0: cpu: 2 mem: 5Gi
NUMA 1: cpu: 5 mem 3 Gi

When we specify the resource request for the deployment we request least resource across all zones for each resource, meaning we would like to request cpu=2 and mem=3 Gi. This is how we can be 100% sure that the resource request would be fulfilled for both the replicas on the their corresponding NUMA nodes.
Hope this helps.